--- Scripts/gen_rdkit_stubs/__init__.py
+++ Scripts/gen_rdkit_stubs/__init__.py
@@ -153,7 +153,11 @@ def clear_stubs(outer_dir):
         outer_dir (_type_): _description_
     """
     for entry in os.listdir(outer_dir):
-        if entry == "CMakeLists.txt":
+        if entry in (
+            "CMakeLists.txt",
+            "gen_rdkit_stubs.out",
+            "gen_rdkit_stubs.err"
+        ):
             continue
         entry = os.path.join(outer_dir, entry)
         if os.path.isdir(entry):
